src/lwt/opentelemetry_lwt.ml

```diff
@@ -29,8 +29,8 @@ module Trace = struct
       Lwt.return rv
     with e ->
       let bt = Printexc.get_raw_backtrace () in
-      let () = finally (Error (Printexc.to_string e, bt)) in
-      Lwt.fail e
+      let () = finally (Error (e, bt)) in
+      raise e
```

Comment:
https://github.com/ocsigen/lwt/blob/48abed72467ca7479e95f1be06d02f40c7c434bd/src/core/lwt.mli#L453-L468

Reply:
I think an even better version would be to use `Lwt.reraise` https://github.com/ocsigen/lwt/blob/48abed72467ca7479e95f1be06d02f40c7c434bd/src/core/lwt.mli#L581-L589 but it would force to use a recent (currently the latest) version of `lwt`

`Lwt.reaise` is just `external reraise : exn -> 'a = "%reraise"` so perhaps we can redefine it in `ocaml-opentelemetry`. WDYT ?

Reply:
sure, that makes sense I think.